_delphi_utils_python/tests/test_archive.py

```diff
@@ -22,125 +22,199 @@
-    "csv4": pd.DataFrame({
-        "geo_id": ["1"],
-        "val": [1.0],
-        "se": [0.1],
-        "sample_size": [10.0]
-    }),
-
-    # Same as 1, but no missing columns ("old-style" file)
-    "csv5": pd.DataFrame({
-        "geo_id": ["1", "2", "3"],
-        "val": [1.0, 2.0, 3.0],
-        "se": [np.nan, 0.20000002, 0.30000003],
-        "sample_size": [10.0, 20.0, 30.0]
-    })
-}
-
-CSVS_AFTER = {
-    # All rows unchanged
-    "csv0": pd.DataFrame({
-        "geo_id": ["1", "2", "3"],
-        "val": [1.0, 2.0, 3.0],
-        "se": [0.10000001, 0.20000002, 0.30000003],
-        "sample_size": [10.0, 20.0, 30.0],
-        "missing_val": [Nans.NOT_MISSING] * 3,
-        "missing_se": [Nans.NOT_MISSING] * 3,
-        "missing_sample_size": [Nans.NOT_MISSING] * 3,
-    }),
-
-    # One row deleted and one row added
-    "csv1": pd.DataFrame({
-        "geo_id": ["1", "2", "4"],
-        "val": [1.0, 2.1, 4.0],
-        "se": [np.nan, 0.21, np.nan],
-        "sample_size": [10.0, 21.0, 40.0],
-        "missing_val": [Nans.NOT_MISSING] * 3,
-        "missing_se": [Nans.CENSORED] + [Nans.NOT_MISSING] * 2,
-        "missing_sample_size": [Nans.NOT_MISSING] * 3,
-    }),
-
-    # File added
-    "csv3": pd.DataFrame({
-        "geo_id": ["2"],
-        "val": [2.0000002],
-        "se": [0.2],
-        "sample_size": [20.0],
-        "missing_val": [Nans.NOT_MISSING],
-        "missing_se": [Nans.NOT_MISSING],
-        "missing_sample_size": [Nans.NOT_MISSING],
-    }),
-
-    # All rows common, but missing columns added
-    "csv4": pd.DataFrame({
-        "geo_id": ["1"],
-        "val": [1.0],
-        "se": [0.1],
-        "sample_size": [10.0],
-        "missing_val": [Nans.NOT_MISSING],
-        "missing_se": [Nans.NOT_MISSING],
-        "missing_sample_size": [Nans.NOT_MISSING],
-    }),
-
-    # Row 1 same, row 2 deleted, row 3 added, row 4 deleted previously
-    # and present again as nan, row 5 deleted previously and absent from file
-    # (no missing columns)
-    "csv5": pd.DataFrame({
-        "geo_id": ["1", "2", "4"],
-        "val": [1.0, 2.1, 4.0],
-        "se": [np.nan, 0.21, np.nan],
-        "sample_size": [10.0, 21.0, 40.0]
-    })
+class Example:
+    def __init__(self, before, after, diff):
+        def fix_df(df):
+            if isinstance(df, pd.DataFrame):
+                return Example._set_df_datatypes(df, CSV_DTYPES)
+            return df
+        self.before = fix_df(before)
+        self.after = fix_df(after)
+        self.diff = fix_df(diff)
+    @staticmethod
+    def _set_df_datatypes(df: pd.DataFrame, dtypes: Dict[str, Any]) -> pd.DataFrame:
+        df = df.copy()
+        for k, v in dtypes.items():
+            if k in df.columns:
+                df[k] = df[k].astype(v)
+        return df
+
+@dataclass
+class Expecteds:
+    deleted: Any
+    common_diffs: Any
+    new: Any
+    raw_exports: Any = None
+    diffed_exports: Any = None
+    filtered_exports: Any = None
+
```

Comment:
Just this change should be good enough

```suggestion
    def __post_init__(self):
        self.raw_exports = list(self.common_diffs.keys()) + self.new
        self.diffed_exports = self.raw_exports + [diff_name for diff_name in self.common_diffs.values() if diff_name is not None]
        self.filtered_exports = [f.replace(".diff", "") for f in self.diffed_exports if f.endswith(".diff")] + self.new
```